heron/instance/src/java/com/twitter/heron/instance/spout/SpoutInstance.java

```diff
@@ -337,6 +341,10 @@ protected void produceTuple() {
       if (currentTime - startOfCycle - instanceEmitBatchTime.toNanos() > 0) {
         break;
       }
+      if (!ByteAmount.fromBytes(newTotalBytesEmitted - totalBytesEmitted)
```

Comment:
do we need to add the same logic in c++ and python?

Reply:
@nwangtw Good catch!
It looks like python already does this but c++ doesn't. I've added that logic as well.